src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx

```diff
@@ -102,14 +106,45 @@ class DatePicker extends React.PureComponent {
         }
     };
 
+    renderDurationPickerInputField = (value) => (
+        <InputField
+            className='datepicker__input'
+            data-tip={false}
+            data-value={value || undefined}
+            error_messages={this.props.validation_errors}
+            is_read_only={this.props.is_read_only}
+            name={this.props.name}
+            onClick={this.handleVisibility}
+            onChange={this.onChangeInput} // TODO: handle duration == days
+            placeholder={this.props.placeholder || localize('Select a duration')}
+            type='number'
+            value={value}
+        />
+    );
+
+    renderDatePickerInputField = (value) => (
+        <InputField
+            className='datepicker__input'
+            data-tip={false}
+            data-value={value || undefined}
+            error_messages={this.props.validation_errors}
+            is_read_only
+            name={this.props.name}
+            onClick={this.handleVisibility}
+            placeholder={this.props.placeholder || localize('Select date')}
+            type='text'
+            value={toMoment(value).format('DD MMM YYYY')}
+        />
+    );
```

Comment:
What about adding a `input_field_props` object shared between `renderDurationPickerInputField` and `renderDatePickerInputField `  to reduce the props duplication?

```javascript
const inputFieldProps = {
  className: `datepicker__input`,
  data-tip: false,
  etc....
}

// and then
renderDurationPickerInputField = (value) => (
<InputField 
 {...inputFieldProps}
/>


renderDatePickerInputField = (value) => (
<InputField 
 {...inputFieldProps}
/>
```